exercises/react/react_test.py

```diff
@@ -1,135 +1,148 @@
-        computeCell2 = reactor.create_compute_cell({inputCell1}, decrement)
-        computeCell3 = reactor.create_compute_cell({computeCell2}, decrement)
-        computeCell4 = reactor.create_compute_cell({computeCell1,
-                                                    computeCell3},
-                                                   product)
-
-        CallbackManager.reset()
-        computeCell4.add_watcher(CallbackManager.count)
-
-        inputCell1.set_value(3)
-        self.assertEqual(CallbackManager.counter, 1)
+from react import InputCell, ComputeCell, Callback
+
+
+# Tests adapted from `problem-specifications//canonical-data.json` @ v1.1.0
+
+class ReactTests(unittest.TestCase):
+    def test_input_cells_have_a_value(self):
+        input_ = InputCell(10)
+        self.assertEqual(input_.value, 10)
+
+    def test_can_set_input_cell_value(self):
+        input_ = InputCell(4)
+        input_.value = 20
+        self.assertEqual(input_.value, 20)
+
+    def test_compute_cells_calculate_initial_value(self):
+        input_ = InputCell(1)
+        output = ComputeCell([input_], lambda inputs: inputs[0] + 1)
+        self.assertEqual(output.value, 2)
+
+    def test_compute_cells_take_inputs_in_right_order(self):
+        one = InputCell(1)
+        two = InputCell(2)
+        output = ComputeCell(
+            [one, two],
+            lambda inputs: inputs[0] + inputs[1]*10
+        )
+        self.assertEqual(output.value, 21)
+
+    def test_compute_cells_update_value_when_dependencies_are_changed(self):
+        input_ = InputCell(1)
+        output = ComputeCell([input_], lambda inputs: inputs[0] + 1)
+
+        input_.value = 3
+        self.assertEqual(output.value, 4)
+
+    def test_compute_cells_can_depend_on_other_compute_cells(self):
+        input_ = InputCell(1)
+        times_two = ComputeCell([input_], lambda inputs: inputs[0] * 2)
+        times_thirty = ComputeCell([input_], lambda inputs: inputs[0] * 30)
+        output = ComputeCell(
+            [times_two, times_thirty],
+            lambda inputs: inputs[0] + inputs[1]
+        )
+
+        self.assertEqual(output.value, 32)
+        input_.value = 3
+        self.assertEqual(output.value, 96)
+
+    def test_compute_cells_fire_callbacks(self):
+        input_ = InputCell(1)
+        output = ComputeCell([input_], lambda inputs: inputs[0] + 1)
+        callback1 = Callback()
+
+        output.add_callback(callback1)
+        input_.value = 3
+        self.assertEqual(callback1.values, [4])
+
+    def test_callback_cells_only_fire_on_change(self):
+        input_ = InputCell(1)
+        output = ComputeCell(
+            [input_],
+            lambda inputs: 111 if inputs[0] < 3 else 222
+        )
+        callback1 = Callback()
+
+        output.add_callback(callback1)
+        input_.value = 2
+        self.assertEqual(callback1.values, [])
+        input_.value = 4
+        self.assertEqual(callback1.values, [222])
```

Comment:
I've been thinking about this: I'm not sure that this is the best interpretation of the canonical data, particularly regarding callbacks. As @petertseng [pointed out](https://github.com/exercism/problem-specifications/pull/1194#issuecomment-367780745), callbacks should just be executable code; they shouldn't "remember" anything.

I'll have to spend some time thinking about a better way to do this. If you can come up with something first, feel free to refactor and update this PR.